synapseclient/core/credentials/credential_provider.py

```diff
@@ -209,61 +159,86 @@ def _get_auth_info(self, syn, user_login_args):
             user_login_args.username,
-            None,
-            None,
             os.environ.get(self.ENVIRONMENT_VAR_NAME),
         )
 
 
 class SynapseCredentialsProviderChain(object):
     """
-    Class that has a list of ``SynapseCredentialsProvider`` from which this class attempts to retrieve
-    ``SynapseCredentials``.
+    Class that has a list of
+    [SynapseCredentialsProvider][synapseclient.core.credentials.credential_provider.SynapseCredentialsProvider]
+    from which this class attempts to retrieve
+    [SynapseCredentials][synapseclient.core.credentials.cred_data.SynapseCredentials].
+
+
+    By default this class uses the following providers in this order:
+
+    1. [UserArgsCredentialsProvider][synapseclient.core.credentials.credential_provider.UserArgsCredentialsProvider]
+    2. [EnvironmentVariableCredentialsProvider][synapseclient.core.credentials.credential_provider.EnvironmentVariableCredentialsProvider]
+    3. [ConfigFileCredentialsProvider][synapseclient.core.credentials.credential_provider.ConfigFileCredentialsProvider]
+    4. [AWSParameterStoreCredentialsProvider][synapseclient.core.credentials.credential_provider.AWSParameterStoreCredentialsProvider]
+
+    Attributes:
+        cred_providers: list of
+            ([SynapseCredentialsProvider][synapseclient.core.credentials.credential_provider.SynapseCredentialsProvider])
+            credential providers
     """
 
-    def __init__(self, cred_providers):
-        """
-        :param list[``SynapseCredentialsProvider``] cred_providers: list of credential providers
-        """
+    def __init__(self, cred_providers) -> None:
         self.cred_providers = list(cred_providers)
 
     @tracer.start_as_current_span("SynapseCredentialsProviderChain::get_credentials")
-    def get_credentials(self, syn, user_login_args):
+    def get_credentials(
+        self, syn: "Synapse", user_login_args: Dict[str, str]
+    ) -> Union[SynapseCredentials, None]:
         """
-        Iterates its list of ``SynapseCredentialsProvider`` and returns the first non-None ``SynapseCredential``
-        returned by a provider. If no provider is able to provide a ``SynapseCredential``, returns None.
-        :param ``synapseclient.client.Synapse`` syn:        Synapse client instance
-        :param ``cred_data.UserLoginArgs`` user_login_args: subset of arguments passed during syn.login()
-        :return: `SynapseCredentials` returned by the first non-None provider in its list, None otherwise
+        Iterates its list of
+        [SynapseCredentialsProvider][synapseclient.core.credentials.credential_provider.SynapseCredentialsProvider]
+        and returns the first non-None
+        [SynapseCredentials][synapseclient.core.credentials.cred_data.SynapseCredentials]
+        returned by a provider. If no provider is able to provide a
+        [SynapseCredentials][synapseclient.core.credentials.cred_data.SynapseCredentials],
+        returns None.
+
+        Parameters:
```

Comment:
```suggestion
        Arguments:
```

Reply:
These actually work the same way on the auto docs. But I will update these to be uniform across the project.